### PR TITLE
Add dependabot

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Adds Dependabot configuration file. Tooling and CI are updated monthly, main modules are updated weekly.

### Motivation

Keep dependencies updated.

### CODE REVIEW

File: .github/dependabot.yaml
Comment (dineshg13) [inline .github/dependabot.yaml:8]: Can we manually trigger one run of the bot ?

Comment (mx-psi) [reply .github/dependabot.yaml:8]: Once this is merged we can trigger a run but it's not possible before merging this. For reference, it is a bit hidden, you have to navigate to Insights > Dependency graph > Dependabot > Click on "Last checked ..." > Click on "Check for updates". 

I can trigger a run after we merge this and see if there are any issues.

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**
